controllers/awsmachine_controller.go

```diff
@@ -629,17 +679,25 @@ func (r *AWSMachineReconciler) createInstance(ec2svc services.EC2MachineInterfac
 	return instance, nil
 }
 
-func (r *AWSMachineReconciler) resolveUserData(machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper) ([]byte, error) {
-	userData, err := machineScope.GetRawBootstrapData()
+func (r *AWSMachineReconciler) resolveUserData(machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper, objectStoreSvc services.ObjectStoreInterface) ([]byte, error) {
+	userData, userDataFormat, err := machineScope.GetRawBootstrapDataWithFormat()
 	if err != nil {
 		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "FailedGetBootstrapData", err.Error())
 		return nil, err
 	}
 
-	if !machineScope.UseSecretsManager() {
-		return userData, nil
+	if machineScope.UseSecretsManager(userDataFormat) {
```

Comment:
Feels like these tests for SSM/ignition and subsequent userdata functions should be abstracted away somehow. Not a requirement for this change but feels like something we should consider for the future.